elks/arch/i86/kernel/irq.c

```diff
@@ -91,6 +128,9 @@ int request_irq(int irq, void (*handler)(int,struct pt_regs *,void *), void *dev
     action->handler = handler;
     action->dev_id = dev_id;
 
+	if (int_handler_add (irq, irq + ((irq > 8) ? 0x68 : 0x08)))
```

Comment:
Passing hard-coded low-core interrupt table addresses re-introduces architecture-specific into this routine, and we just removed all that. (for this call and system call 0x80 below). IMO, `int_handler_add` should only be passed a single parameter - `irq`, and `int_vector_set` should handle architecture-specific low-memory interrupt address from passed irq.

Reply:
Agree, the `irq-8859.c` should provide an IRQ to vector mapping, as it already provides the IRQ remapping.

Reply:
Done in cbe3e07